Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.h

```diff
@@ -77,11 +76,20 @@ namespace AzToolsFramework
         bool m_expanded = true;
         QCheckBox* m_expanderWidget = nullptr;
 
-        //! Vector containing vectors of widgets, where each vector represents a unique shared column group.
-        //! Each widget in a vector will be a widget that belongs to that shared column group.
-        AZStd::vector<AZStd::vector<size_t>> m_sharePriorColumn;
+        // Contains the indices of all widgets that start a new column.
+        AZStd::unordered_set<size_t> m_columnStarts;
 
     private:
+        void CloseColumn(
+            QHBoxLayout* currentColumnLayout,
+            QRect& itemGeometry,
+            int& currentColumnCount,
+            const int columnWidth,
```

Comment:
nit: Don't really need a `const` here, unless you really want to prevent this function from modifying the parameter value inside of the function,

Reply:
I'm gonna touch this function in the next PR, I'll fix it then.